crates/viewer/re_space_view_spatial/src/visualizers/utilities/proc_mesh_vis.rs

```diff
@@ -0,0 +1,316 @@
+        // TODO(#7026): We should formalize this kind of hybrid joining better.
+        let num_instances = batch
+            .half_sizes
+            .len()
+            .max(ent_context.transform_info.reference_from_instances.len());
+        let half_sizes = clamped_or_nothing(batch.half_sizes, num_instances);
+
+        let (annotation_infos, _) = process_annotation_and_keypoint_slices(
+            self.query.latest_at,
+            num_instances,
+            std::iter::repeat(glam::Vec3::ZERO).take(num_instances),
+            batch.keypoint_ids,
+            batch.class_ids,
+            &ent_context.annotations,
+        );
+
+        // Has not custom fallback for radius, so we use the default.
+        // TODO(andreas): It would be nice to have this handle this fallback as part of the query.
+        let line_radii = process_radius_slice(
+            entity_path,
+            num_instances,
+            batch.line_radii,
+            components::Radius::default(),
+        );
+        let colors = process_color_slice(
+            query_context,
+            self.fallback,
+            num_instances,
+            &annotation_infos,
+            batch.colors,
+        );
+
+        let mut line_batch = self
+            .line_builder
+            .batch(self.line_batch_debug_label.clone())
+            .depth_offset(ent_context.depth_offset)
+            .outline_mask_ids(ent_context.highlight.overall)
+            .picking_object_id(re_renderer::PickingLayerObjectId(entity_path.hash64()));
+
+        let mut world_space_bounding_box = re_math::BoundingBox::NOTHING;
+
+        let world_from_instances = ent_context
+            .transform_info
+            .clamped_reference_from_instances();
+
+        let mut num_instances = 0;
+        for (
+            instance_index,
+            (half_size, world_from_instance, radius, &color, proc_mesh_key, fill_mode),
+        ) in itertools::izip!(
+            half_sizes,
+            world_from_instances,
+            line_radii,
+            colors.iter(),
+            batch.meshes,
+            batch.fill_modes
+        )
+        .enumerate()
+        {
+            let instance = Instance::from(instance_index as u64);
+            num_instances = instance_index + 1;
+
+            let world_from_instance = world_from_instance
+                * glam::Affine3A::from_scale(glam::Vec3::from(*half_size))
+                * constant_instance_transform;
+            world_space_bounding_box = world_space_bounding_box.union(
+                proc_mesh_key
+                    .simple_bounding_box()
+                    .transform_affine3(&world_from_instance),
+            );
+
+            match fill_mode {
+                FillMode::MajorWireframe | FillMode::DenseWireframe
+                    if proc_mesh_key == ProcMeshKey::Cube =>
+                {
+                    // TODO(kpreid): Does this special case for cubes have any value?
```

Comment:
agreed, would be nice to further simplify this at some point and reduce complexity on the line batch
(and I imagine the less complex the ad-hoc line machinery becomes the easier it might get later to move to more retained & instancing based model)

Reply:
I'll consider that for the future, and not make this PR hairier by changing the size here.

Reply:
…whoops, that reply was meant for the thread about the 0.5 size.

Reply:
Done.